### PR TITLE
refactor(amplify-cli): convert require()s to import()s

### PR DESCRIPTION
**Pull request by jhockett**
#### Description of changes
 - Remove dynamic require()s where possible in the `amplify-cli-internal` package
 - Converted a number of `.forEach` loops to `for...of` for simpler `await` usage

#### Description of how you validated changes
`yarn test`, e2e passes

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [x] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sobolk — APPROVED** [review]

File: packages/amplify-cli/src/execution-manager.ts [hunk 1]
```diff
@@ -161,7 +161,7 @@ const executePluginModuleCommand = async (context: Context, plugin: PluginInfo):
   }
 
   const handler = await getHandler(plugin, context);
-  attachContextExtensions(context, plugin);
+  await attachContextExtensions(context, plugin);
```
Comment (evcodes) [inline packages/amplify-cli/src/execution-manager.ts:164]: why are we awaiting here?

File: packages/amplify-cli/src/init-steps/s0-analyzeProject.ts [hunk 2]
```diff
@@ -76,18 +76,17 @@ const displayAndSetDefaults = async (context: $TSContext, projectPath: string, p
   context.print.success('The following configuration will be applied:');
   context.print.info('');
 
-  await displayConfigurationDefaults(context, defaultProjectName, defaultEnv, defaultEditorName);
+  displayConfigurationDefaults(context, defaultProjectName, defaultEnv, defaultEditorName);
```
Comment (evcodes) [inline packages/amplify-cli/src/init-steps/s0-analyzeProject.ts:79]: and not awaiting here?

Comment (jhockett) [reply packages/amplify-cli/src/init-steps/s0-analyzeProject.ts:79]: That await was unnecessary, the function is not async

File: packages/amplify-cli/src/execution-manager.ts (hunk 1 above)
Comment (jhockett) [reply packages/amplify-cli/src/execution-manager.ts:164]: I converted the function to be async

**Review by akshbhu — APPROVED** [review]

**Pull request merged**